### ISSUE TITLE
Deserialization issue from XML to POJO with Jackson: no String-argument constructor/factory method to deserialize from String value

### ISSUE THREAD

**Issue by mvezelis-stunn** [FasterXML/jackson-dataformat-xml#219]
I've got an XML document which I need to convert (deserialize) into Java POJO. I use Java 8, and Jackson framework for mapping.
 **Gradle** dependencies:

```
dependencies {
    compile('com.fasterxml.jackson.dataformat:jackson-dataformat-xml')
    compile('org.springframework.boot:spring-boot-starter-freemarker')
    compile('org.springframework.boot:spring-boot-starter-web')
    providedRuntime('org.springframework.boot:spring-boot-starter-tomcat')
    testCompile('org.springframework.boot:spring-boot-starter-test')
}
```
Wrapper XML Doc:

```
@JacksonXmlRootElement(localName = "rates_response")
public class RatesResponse implements Serializable{
    private static final long serialVersionUID = 3254688495454519L;

    @JacksonXmlProperty(isAttribute = true)
    private String b_number = "";

    @JacksonXmlProperty(isAttribute = true)
    private String l_premium = "";

    @JacksonXmlProperty(isAttribute = true)
    private String currency = "";

    @XmlElement(required = false)
    @JacksonXmlProperty
    private String c_b_relationship = null;

    @JacksonXmlProperty(isAttribute = true)
    private String n_of_p_loans = null;

    /*
     * Key: status_code
     * Key: status_explanation
     */
    @JacksonXmlElementWrapper
    private Map<String, String> message_status = null;

   // @XmlElement
    @JacksonXmlElementWrapper(useWrapping = false)
    private List<RateDetails> rate_details = null;

    public RatesResponse(){}
     //GETTERS/SETTERS
```
**Child document** (List<RateDetails> rate_details = null;):

```
@JacksonXmlRootElement(localName = "rate_details")
public class RateDetails implements Serializable {
    private static final long serialVersionUID = 324564684815158619L;

    @JacksonXmlProperty
    private String r_reference = null;
    @JacksonXmlProperty
    private String n_of_p_days = null;
    @JacksonXmlProperty
    private String n_of_instalments = null;
    @JacksonXmlProperty
    private String n_rate = null;
    @JacksonXmlProperty
    private String p_p_available = null;

    @XmlElement(required = false)
    @JacksonXmlProperty
    private String i_overrider = null;//not always appears in doc
    @JacksonXmlProperty
    private String a_fee = null;
    @JacksonXmlProperty
    private String m_fee = null;
    @JacksonXmlProperty
    private String t_fees = null;
    @JacksonXmlProperty
    private String t_c_for_credit = null;
    @JacksonXmlProperty
    private String d_amount = null;
    @JacksonXmlProperty
    private String apr = null;

    public RateDetails(){}
    //GETTERS/SETTERS
```
XML document to parse:

```
<?xml version="1.0"?>
<!DOCTYPE rates_response SYSTEM "p_f_interface.dtd">
<rates_response b_number="KK5000000000" l_premium="5000.00" currency="GBP" >
    <message_status>
        <status_code>1000</status_code>
        <status_explanation>Success</status_explanation>
    </message_status>
    <c_b_relationship n_of_p_loans="0"/>
    <rate_details>
        <r_reference>AS28</r_reference>
        <n_of_p_days>28</n_of_p_days>
        <n_of_instalments>12</n_of_instalments>
        <n_rate>7.00</n_rate>
        <p_p_available>NO</p_p_available>
        <i_overrider>3.65</i_overrider>
        <a_fee>0.00</a_fee>
        <m_fee>0.00</m_fee>
        <t_fees>0.00</t_fees>
        <t_c_for_credit>350.00</t_c_for_credit>
        <d_amount>0.00</d_amount>
        <apr>16.05</apr>
    </rate_details>    
</rates_response>
```
The method which parses the doc:

```
private RatesResponse readRatesDoc() {
    RatesResponse xmlContainer = null;

    ObjectMapper xmlMapper = new XmlMapper();
    try {
        xmlContainer = xmlMapper.readValue(contrUtil.readXmlFile("Rates_Response.xml"), RatesResponse.class);
    } catch (IOException e) {
        log.log(Level.SEVERE, "An Exception on parsing XML document", e);
    }
    return xmlContainer;
}
```
The exception:

```
com.fasterxml.jackson.databind.JsonMappingException: Can not construct instance of premium.finance.beanResponse.rates.RateDetails: no String-argument constructor/factory method to deserialize from String value ('16.05')
 at [Source: java.io.StringReader@7bd22d0b; line: 1, column: 945] (through reference chain: premium.finance.beanResponse.rates.RatesResponse["rate_details"]->java.util.ArrayList[1])
    at com.fasterxml.jackson.databind.JsonMappingException.from(JsonMappingException.java:261) ~[jackson-databind-2.8.3.jar:2.8.3]
    at com.fasterxml.jackson.databind.DeserializationContext.instantiationException(DeserializationContext.java:1456) ~[jackson-databind-2.8.3.jar:2.8.3]
    at com.fasterxml.jackson.databind.DeserializationContext.handleMissingInstantiator(DeserializationContext.java:1012) ~[jackson-databind-2.8.3.jar:2.8.3]
    at com.fasterxml.jackson.databind.deser.ValueInstantiator._createFromStringFallbacks(ValueInstantiator.java:366) ~[jackson-databind-2.8.3.jar:2.8.3]
    at com.fasterxml.jackson.databind.deser.std.StdValueInstantiator.createFromString(StdValueInstantiator.java:307) ~[jackson-databind-2.8.3.jar:2.8.3]
    at com.fasterxml.jackson.databind.deser.BeanDeserializerBase.deserializeFromString(BeanDeserializerBase.java:1278) ~[jackson-databind-2.8.3.jar:2.8.3]
    at com.fasterxml.jackson.databind.deser.BeanDeserializer._deserializeOther(BeanDeserializer.java:159) ~[jackson-databind-2.8.3.jar:2.8.3]
    at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserialize(BeanDeserializer.java:150) ~[jackson-databind-2.8.3.jar:2.8.3]
    at com.fasterxml.jackson.databind.deser.std.CollectionDeserializer.deserialize(CollectionDeserializer.java:277) ~[jackson-databind-2.8.3.jar:2.8.3]
    at com.fasterxml.jackson.databind.deser.std.CollectionDeserializer.deserialize(CollectionDeserializer.java:249) ~[jackson-databind-2.8.3.jar:2.8.3]
    at com.fasterxml.jackson.databind.deser.std.CollectionDeserializer.deserialize(CollectionDeserializer.java:26) ~[jackson-databind-2.8.3.jar:2.8.3]
    at com.fasterxml.jackson.databind.deser.SettableBeanProperty.deserialize(SettableBeanProperty.java:499) ~[jackson-databind-2.8.3.jar:2.8.3]
    at com.fasterxml.jackson.databind.deser.impl.MethodProperty.deserializeAndSet(MethodProperty.java:101) ~[jackson-databind-2.8.3.jar:2.8.3]
    at com.fasterxml.jackson.databind.deser.BeanDeserializer.vanillaDeserialize(BeanDeserializer.java:276) ~[jackson-databind-2.8.3.jar:2.8.3]
    at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserialize(BeanDeserializer.java:140) ~[jackson-databind-2.8.3.jar:2.8.3]
    at com.fasterxml.jackson.dataformat.xml.deser.WrapperHandlingDeserializer.deserialize(WrapperHandlingDeserializer.java:120) ~[jackson-dataformat-xml-2.8.3.jar:2.8.3]
    at com.fasterxml.jackson.databind.ObjectMapper._readMapAndClose(ObjectMapper.java:3789) ~[jackson-databind-2.8.3.jar:2.8.3]
    at com.fasterxml.jackson.databind.ObjectMapper.readValue(ObjectMapper.java:2833) ~[jackson-databind-2.8.3.jar:2.8.3]
    at premium.finance.controller.IndexController.readRatesDoc(IndexController.java:190) [main/:na]
    at premium.finance.controller.IndexController.requestIndex(IndexController.java:69) [main/:na]
    at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[na:1.8.0_91]
    at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[na:1.8.0_91]
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[na:1.8.0_91]
    at java.lang.reflect.Method.invoke(Method.java:498) ~[na:1.8.0_91]
    at org.springframework.web.method.support.InvocableHandlerMethod.doInvoke(InvocableHandlerMethod.java:221) [spring-web-4.3.3.RELEASE.jar:4.3.3.RELEASE]
    at org.springframework.web.method.support.InvocableHandlerMethod.invokeForRequest(InvocableHandlerMethod.java:136) [spring-web-4.3.3.RELEASE.jar:4.3.3.RELEASE]
    at org.springframework.web.servlet.mvc.method.annotation.ServletInvocableHandlerMethod.invokeAndHandle(ServletInvocableHandlerMethod.java:114) [spring-webmvc-4.3.3.RELEASE.jar:4.3.3.RELEASE]
    at org.springframework.web.servlet.mvc.method.annotation.RequestMappingHandlerAdapter.invokeHandlerMethod(RequestMappingHandlerAdapter.java:827) [spring-webmvc-4.3.3.RELEASE.jar:4.3.3.RELEASE]
    at org.springframework.web.servlet.mvc.method.annotation.RequestMappingHandlerAdapter.handleInternal(RequestMappingHandlerAdapter.java:738) [spring-webmvc-4.3.3.RELEASE.jar:4.3.3.RELEASE]
    at org.springframework.web.servlet.mvc.method.AbstractHandlerMethodAdapter.handle(AbstractHandlerMethodAdapter.java:85) [spring-webmvc-4.3.3.RELEASE.jar:4.3.3.RELEASE]
    at org.springframework.web.servlet.DispatcherServlet.doDispatch(DispatcherServlet.java:963) [spring-webmvc-4.3.3.RELEASE.jar:4.3.3.RELEASE]
    at org.springframework.web.servlet.DispatcherServlet.doService(DispatcherServlet.java:897) [spring-webmvc-4.3.3.RELEASE.jar:4.3.3.RELEASE]
    at org.springframework.web.servlet.FrameworkServlet.processRequest(FrameworkServlet.java:970) [spring-webmvc-4.3.3.RELEASE.jar:4.3.3.RELEASE]
    at org.springframework.web.servlet.FrameworkServlet.doGet(FrameworkServlet.java:861) [spring-webmvc-4.3.3.RELEASE.jar:4.3.3.RELEASE]
    at javax.servlet.http.HttpServlet.service(HttpServlet.java:622) [tomcat-embed-core-8.5.5.jar:8.5.5]
    at org.springframework.web.servlet.FrameworkServlet.service(FrameworkServlet.java:846) [spring-webmvc-4.3.3.RELEASE.jar:4.3.3.RELEASE]
    at javax.servlet.http.HttpServlet.service(HttpServlet.java:729) [tomcat-embed-core-8.5.5.jar:8.5.5]
    at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:230) [tomcat-embed-core-8.5.5.jar:8.5.5]
    at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:165) [tomcat-embed-core-8.5.5.jar:8.5.5]
    at org.apache.tomcat.websocket.server.WsFilter.doFilter(WsFilter.java:52) [tomcat-embed-websocket-8.5.5.jar:8.5.5]
    at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:192) [tomcat-embed-core-8.5.5.jar:8.5.5]
    at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:165) [tomcat-embed-core-8.5.5.jar:8.5.5]
    at org.springframework.web.filter.RequestContextFilter.doFilterInternal(RequestContextFilter.java:99) [spring-web-4.3.3.RELEASE.jar:4.3.3.RELEASE]
    at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:107) [spring-web-4.3.3.RELEASE.jar:4.3.3.RELEASE]
    at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:192) [tomcat-embed-core-8.5.5.jar:8.5.5]
    at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:165) [tomcat-embed-core-8.5.5.jar:8.5.5]
    at org.springframework.web.filter.HttpPutFormContentFilter.doFilterInternal(HttpPutFormContentFilter.java:89) [spring-web-4.3.3.RELEASE.jar:4.3.3.RELEASE]
    at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:107) [spring-web-4.3.3.RELEASE.jar:4.3.3.RELEASE]
    at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:192) [tomcat-embed-core-8.5.5.jar:8.5.5]
    at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:165) [tomcat-embed-core-8.5.5.jar:8.5.5]
    at org.springframework.web.filter.HiddenHttpMethodFilter.doFilterInternal(HiddenHttpMethodFilter.java:77) [spring-web-4.3.3.RELEASE.jar:4.3.3.RELEASE]
    at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:107) [spring-web-4.3.3.RELEASE.jar:4.3.3.RELEASE]
    at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:192) [tomcat-embed-core-8.5.5.jar:8.5.5]
    at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:165) [tomcat-embed-core-8.5.5.jar:8.5.5]
    at org.springframework.web.filter.CharacterEncodingFilter.doFilterInternal(CharacterEncodingFilter.java:197) [spring-web-4.3.3.RELEASE.jar:4.3.3.RELEASE]
    at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:107) [spring-web-4.3.3.RELEASE.jar:4.3.3.RELEASE]
    at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:192) [tomcat-embed-core-8.5.5.jar:8.5.5]
    at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:165) [tomcat-embed-core-8.5.5.jar:8.5.5]
    at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:198) [tomcat-embed-core-8.5.5.jar:8.5.5]
    at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:108) [tomcat-embed-core-8.5.5.jar:8.5.5]
    at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:472) [tomcat-embed-core-8.5.5.jar:8.5.5]
    at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:140) [tomcat-embed-core-8.5.5.jar:8.5.5]
    at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:79) [tomcat-embed-core-8.5.5.jar:8.5.5]
    at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:87) [tomcat-embed-core-8.5.5.jar:8.5.5]
    at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:349) [tomcat-embed-core-8.5.5.jar:8.5.5]
    at org.apache.coyote.http11.Http11Processor.service(Http11Processor.java:784) [tomcat-embed-core-8.5.5.jar:8.5.5]
    at org.apache.coyote.AbstractProcessorLight.process(AbstractProcessorLight.java:66) [tomcat-embed-core-8.5.5.jar:8.5.5]
    at org.apache.coyote.AbstractProtocol$ConnectionHandler.process(AbstractProtocol.java:802) [tomcat-embed-core-8.5.5.jar:8.5.5]
    at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.doRun(NioEndpoint.java:1410) [tomcat-embed-core-8.5.5.jar:8.5.5]
    at org.apache.tomcat.util.net.SocketProcessorBase.run(SocketProcessorBase.java:49) [tomcat-embed-core-8.5.5.jar:8.5.5]
    at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142) [na:1.8.0_91]
    at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617) [na:1.8.0_91]
    at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:61) [tomcat-embed-core-8.5.5.jar:8.5.5]
    at java.lang.Thread.run(Thread.java:745) [na:1.8.0_91]
```
The top line says:

`no String-argument constructor/factory method to deserialize from String value ('16.05')`

The value it is referring to comes from under this tag:
`<apr>16.05</apr>`

If I displace (move around) the tag, <apr>16.05</apr>, the parser works and no exception gets caused. If I replace the tag by different tag, the same exception still occurs; **it affects only the last tag within the <rate_details> wrapper tag**, for some reasons.

**Comment by cowtowncoder** [comment]
And this is with a modern version of Jackson (like 2.7.9 or 2.8.6)?

**Comment by mvezelis-stunn** [comment]
Hi,
The dependencies that I use have been displayed at Gradle build file (included with the report).

In fact, I performed parsing of the XML file into Java POJO with “rest template”, which comes with Spring framework. So, all is good, I get the POJO. 
But if I switch to Jackson parser I get that error reported with the issue. So far, I get the error only parsing to POJO on that specific Java bean obj.

Best regards,
Mr Mindaugas Vezelis

**Comment by tomcalado** [comment]
I have the same problem. For me, it works using the follow code:

`JacksonXmlModule module = new JacksonXmlModule();`
`module.setDefaultUseWrapper(false);`
`XmlMapper xmlMapper = new XmlMapper(module);`

instead of your line that is:

`ObjectMapper xmlMapper = new XmlMapper();`

**Comment by lex-em** [comment]
Problem reproduced in 2.8.7

**Comment by lex-em** [comment]
2.4.*, 2.5.*, 2.6.*, 2.7.* versions affected too

**Comment by Shooting1007** [comment]
I havathe same problem. Is there any solution to solve?

**Comment by glucy** [comment]
Seeing the same issue in Jackson 2.8.9

**Comment by lex-em** [comment]
does it fixed? release 2.9 is sheduled?

**Comment by yvzhelnin** [comment]
Reproduced in 2.9.0.pr4 when I was attemting to parse https://news.rambler.ru/rss/head/

**Comment by lex-em** [comment]
I get it work in many versions, in TC case just remove JacksonXmlRootElement from child RateDetails class

**Comment by evgzakharov** [comment]
possible solution to fix this bug:
change `@JacksonXmlElementWrapper` to `@JacksonXmlProperty`
 
```java
@JacksonXmlProperty
private List<RateDetails> rate_details = null;
```

**Comment by cowtowncoder** [comment]
Possibly related: #254

**Comment by sushil5612** [comment]
Is it fixed. I am facing the same problem

**Comment by haroonismailbasha** [comment]
using @JacksonXmlProperty in 2.8.7 solved the problem. it took long time as it was giving null
Example element value is given below

import com.fasterxml.jackson.dataformat.xml.annotation.JacksonXmlProperty;
@JacksonXmlProperty(localName = "Value")
    private String Value;

    public String getValue ()
    {
        return Value;
    }

    public void setValue (String Value)
    {
        this.Value = Value;
    }

**Comment by johnzielke** [comment]
Are there any updates on this? I'm having the same issue

**Comment by viktorkondin** [comment]
Have the same issue with 2.9.8 and 2.9.9

**Comment by cowtowncoder** [comment]
At this point I would need a simplified version of remaining problem, verified against one of `2.10.0` pre-releases (like `2.10.0.pr3`). Fix, if any, would not be backported in 2.9 or earlier versions, at this point, but would be part of 2.10.0 releases (official 2.10.0 is not yet out ,will be by end of Sep 2019).

Verifying against 2.10 is important since there have been a few fixes, and improvements to error reporting; and also because longer discussions here suggest that some definitions work and others may be not -- original example is so long that it is difficult to test against. But it should be possible to simplify it significantly, if anyone has time to help here.

**Comment by viktorkondin** [comment]
You can easily reproduce it with next conditions:
1) Jackson 2.9.8 or 2.9.9
2) Any xml document, which has attribute and property after it

Example:
`<worker>
<developer>test1</developer>
<tester grade="senior">test2</tester>
<manager>test3</manager>
</worker>`

If I create class like:
`public class Worker
{
private String developer;
private String tester;
private String manager;

// setters and getters
}`

It will fail during de-serialization on manager step / line. For developers it might not be obvious, that actual issue in attribute "grade" from previous property, which was explicitly ignored.

Workaround to solve this issue is to create next 2 classes:
`public class Worker
{
private String developer;
private Tester tester;
private String manager;

// setters and getters
}`

`public class Tester
{
private String grade;
private String value;  // ??? still dont understand how and why it works ...

// setters and getters
}`

**Comment by cowtowncoder** [comment]
Thank you. Yes, this is problematic as attributes are not ignored/ignorable by default -- there will essentially be 2 pieces of information, and this can not be readily mapped to a String.
I can see how logically it would be good to allow attributes to be essentially ignored in special case of a scalar target value (like `String`), but the trick is that of how to do it through streaming interface.

**Comment by viktorkondin** [comment]
May be it would make sense to fail de-serialization on property with attribute, which was declared as String / Integer, while custom object was actually expected? It should not affect existing projects anyway.

**Comment by cowtowncoder** [comment]
I think there are couple of ways that would make sense, but the challenge is how and where to figure out failure mode (or support handling)... since the information on something coming from an attribute is sort of lost by the time token stream is available, and since databind-level deserializers have no concept of alternate data formats.

But I agree, even just improved error handling would be valuable, to let user know what the problem is.

**Comment by Pytry** [comment]
I came across this error, but for me it wasn't a bug; it was a bad DTO.

**Simplified xml example**

    <line>
        <code type="ABC">qsd</code>
        <amount>138</amount>
    </line>

**Simplified DTO Example**

    @JacksonXmlRootElement(localName="line")
    public class Line{
        @JacksonXmlProperty(localName="code")
        private String code; //This should have been a complex type!
        @JacksonXmlProperty(localName="amount")
        private String amount;
        // getters and setters
    }

The error message in this case was confusing because it made me think the problem was with 
`<amount>`, when it was actually for `<code>`.

**Comment by madisoncool** [comment]
Is there any workaround for this issue?

**Comment by cowtowncoder** [comment]
@Pytry yes, the misleading error message is the problem here. In addition it would probably also make sense for Jackson to be able to ignore "unbound" attribute values in cases where element is bound to scalar value, but this is unfortunately much more complicated thing to achieve (as opposed to general "just ignore unknown things").

**Comment by zhiweipeng** [comment]
I also ran into same issue, the stack trace is almost same as what's originally reported in this issue. And I noticed one thing in common is **CollectionDeserializer** (in POJO an ArrayList is used), my guess is it's unable to find the correct boundary of the bean (which is in the ArrayList), so ended up using a simple string to deserialize into the bean, but I am not familiar with Jackson code so I can be wrong. 

I also tried all the suggestions in this thread but none of them worked. So even though the error message is very similar but the root cause might be different, and agree with others that this message is misleading.


Caused by: com.fasterxml.jackson.databind.exc.MismatchedInputException: Cannot construct instance of `....response.type.CustSrch` (although at least one Creator exists): no String-argument constructor/factory method to deserialize from String value ('000000002856')
 at [Source: (StringReader); line: 5, column: 28] (through reference chain: ....response.CustSrchInqRs["CustSrch"]->java.util.ArrayList[0])
	at com.fasterxml.jackson.databind.exc.MismatchedInputException.from(MismatchedInputException.java:63)
	at com.fasterxml.jackson.databind.DeserializationContext.reportInputMismatch(DeserializationContext.java:1429)
	at com.fasterxml.jackson.databind.DeserializationContext.handleMissingInstantiator(DeserializationContext.java:1059)
	at com.fasterxml.jackson.databind.deser.ValueInstantiator._createFromStringFallbacks(ValueInstantiator.java:371)
	at com.fasterxml.jackson.databind.deser.std.StdValueInstantiator.createFromString(StdValueInstantiator.java:323)
	at com.fasterxml.jackson.databind.deser.BeanDeserializerBase.deserializeFromString(BeanDeserializerBase.java:1373)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer._deserializeOther(BeanDeserializer.java:171)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserialize(BeanDeserializer.java:161)
	at com.fasterxml.jackson.databind.deser.std.CollectionDeserializer.deserialize(CollectionDeserializer.java:286)
	at com.fasterxml.jackson.databind.deser.std.CollectionDeserializer.deserialize(CollectionDeserializer.java:245)
	at com.fasterxml.jackson.databind.deser.std.CollectionDeserializer.deserialize(CollectionDeserializer.java:27)
	at com.fasterxml.jackson.databind.deser.impl.MethodProperty.deserializeAndSet(MethodProperty.java:129)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.vanillaDeserialize(BeanDeserializer.java:288)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserialize(BeanDeserializer.java:151)
	at com.fasterxml.jackson.databind.ObjectMapper._readMapAndClose(ObjectMapper.java:4202)
	at com.fasterxml.jackson.databind.ObjectMapper.readValue(ObjectMapper.java:3205)
	at com.fasterxml.jackson.databind.ObjectMapper.readValue(ObjectMapper.java:3173)
	... 64 more

**Comment by taghreedHamdyAbdelhakim** [comment]
I hava the same problem. Is there any solution to solve?
i'am using spring boot 2.2.4 RELEASE ,and com.fasterxml.jackson 2.11.0

**Comment by cowtowncoder** [comment]
One quick note for anyone adding comments here: just because exception message looks similar does not mean the underlying problem is the same.

So please only add a note "I have the same problem" if you can add some specific new information (including Java type(s) involved and matching XML if necessary). Or, possibly, file a separate new issue with reference like "looks similar to # 219".
Just saying you see something similar does not really help either fix the problem here or address issue you have.

**Comment by cowtowncoder** [comment]
@viktorkondin @Pytry (un?)fortunately I can not reproduce your issue at this point, with 2.10.4.

I think I will mark this as fixed for 2.10.5 (next release), although it probably was fixed earlier, perhaps in 2.10.0 already.

**Comment by cowtowncoder** [comment]
@prydin status is that this issue is closed.

If you have remaining problems, please file a new issue with full reproduction. This may or may not be the same underlying problem (just getting same exception does not necessarily mean it is) and due to long discussion across various versions it makes sense to start from clean slate. You may add a reference to this issue as background for anyone interested.
